Common/Net/HTTPNaettRequest.cpp

```diff
@@ -0,0 +1,96 @@
+#ifndef HTTPS_NOT_AVAILABLE
+
+#include <cstring>
+
+#include "Common/Net/HTTPRequest.h"
+#include "Common/Net/HTTPNaettRequest.h"
+#include "Common/Thread/ThreadUtil.h"
+#include "Common/Log.h"
+
+#include "ext/naett/naett.h"
+
+namespace http {
+
+HTTPSDownload::HTTPSDownload(RequestMethod method, const std::string &url, const std::string &postData, const std::string &postMime, const Path &outfile, ProgressBarMode progressBarMode, const std::string &name)
+	: Download(url, name, &cancelled_), method_(method), postData_(postData), postMime_(postMime), outfile_(outfile), progressBarMode_(progressBarMode) {
+}
+
+HTTPSDownload::~HTTPSDownload() {
+	Join();
+}
+
+void HTTPSDownload::Start() {
+	_dbg_assert_(!req_);
+
+	const char *methodStr = method_ == RequestMethod::GET ? "GET" : "POST";
+	req_ = naettRequest_va(url_.c_str(), !postMime_.empty() ? 3 : 2, naettMethod(methodStr), naettHeader("accept", "application/json, text/*; q=0.9, */*; q=0.8"), naettHeader("Content-Type", postMime_.c_str()));
+	res_ = naettMake(req_);
```

Comment:
Seems like this is missing any timeout setting, which naett seems to support.

-[Unknown]

Reply:
Added (though will make more configurable in a followup with a RequestDesc struct

Reply:
Yeah, there's `SetDataTimeout()` currently, which I guess Download doesn't currently call.

-[Unknown]